### PR TITLE
feature: stack widget

### LINKED ISSUE #59: Stack widget Logic
Stack (FIFO/LIFO)

### PR DESCRIPTION
**Pull request by D15ND**
1. **Summary:** _Stack widget_

2. **Linked Issue:** Closes #59 

3. **UI Screenshots:**
image
image

---

## Definition of Done (DoD)

Before requesting review, confirm that:

- [x] Acceptance criteria are satisfied
- [x] No debug logs or commented-out code remain
- [x] ESLint and formatting checks pass
- [x] No console errors or warnings
- [x] All existing tests pass locally
- [x] Edge cases and error states are handled
- [x] Pull request is linked to an issue

---

### Code Quality

- [x] My code follows the team's style guide and conventions
- [x] No type assertions are used
- [x] No explicit `any` types are used

### Testing

- [x] I have added or updated tests to cover my changes

### CODE REVIEW

File: frontend/src/components/features/widgets/Stack/Stack.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+              )}
+            </Box>
+          </Box>
+        )}
+      </Box>
+    </Container>
+  );
+};
+
+export default StackWidget;
```
[the remark this answers is not in the document]
Comment (D15ND) [reply frontend/src/components/features/widgets/Stack/Stack.tsx]: Thank you. 
Fixed

File: frontend/src/components/features/widgets/Stack/Stack.test.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { MantineProvider } from '@mantine/core';
+import { describe, it, expect } from 'vitest';
+import StackWidget from './Stack';
+import { QUIZ_QUESTIONS, FEEDBACK_MESSAGES } from './Stack.constants';
+
+const renderWithMantine = (component: React.ReactElement) => {
+  return render(<MantineProvider>{component}</MantineProvider>);
+};
```
[the remark this answers is not in the document]
Comment (D15ND) [reply frontend/src/components/features/widgets/Stack/Stack.test.tsx]: done
